src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -398,17 +403,17 @@ pub struct RSVirtualResult;
 
 /// Holds the actual data of an ['IndexResult']
 #[repr(C)]
-pub union RSIndexResultData<'a> {
-    pub agg: ManuallyDrop<RSAggregateResult<'a>>,
-    pub term: ManuallyDrop<RSTermRecord<'a>>,
+pub union RSIndexResultData<'index, 'aggregate_children> {
+    pub agg: ManuallyDrop<RSAggregateResult<'index, 'aggregate_children>>,
+    pub term: ManuallyDrop<RSTermRecord<'index>>,
     pub num: ManuallyDrop<RSNumericRecord>,
     pub virt: ManuallyDrop<RSVirtualResult>,
 }
 
 /// The result of an inverted index
```

Comment:
Worth documenting the exact semantic of both lifetimes here.

Reply:
Done in https://github.com/RediSearch/RediSearch/pull/6640/commits/865e5cfa3b70b45764788e55cb60f84aa662db0a